pkg/controllers/clusterresourceplacement/controller.go

```diff
@@ -1025,13 +1026,26 @@ func isRolloutCompleted(crp *fleetv1beta1.ClusterResourcePlacement) bool {
 
 	expectedCondTypes := determineExpectedCRPAndResourcePlacementStatusCondType(crp)
 	for _, i := range expectedCondTypes {
-		if !condition.IsConditionStatusTrue(crp.GetCondition(string(i.ClusterResourcePlacementConditionType())), crp.Generation) {
+		cond := crp.GetCondition(string(i.ClusterResourcePlacementConditionType()))
+		if cond != nil {
+			emitPlacementStatusMetric(cond, crp)
+		}
+		if !condition.IsConditionStatusTrue(cond, crp.Generation) {
 			return false
 		}
 	}
 	return true
 }
 
 func isCRPScheduled(crp *fleetv1beta1.ClusterResourcePlacement) bool {
-	return condition.IsConditionStatusTrue(crp.GetCondition(string(fleetv1beta1.ClusterResourcePlacementScheduledConditionType)), crp.Generation)
+	cond := crp.GetCondition(string(fleetv1beta1.ClusterResourcePlacementScheduledConditionType))
+	if cond != nil {
+		emitPlacementStatusMetric(cond, crp)
```

Comment:
there are multiple ways to create the status metrics,

for example, 
the label is, crp name, generation, value is status for that generation.

could you please investigate how others emit metrics?

Reply:
I wonder if including the crp name can result in cardinality being too large?